### PR TITLE
Documented the deprecation of the Templating component

### PR DESCRIPTION
**Pull request by javiereguiluz**
Documents https://github.com/symfony/symfony/pull/21035 but there are some things I'm not sure about. Please, review.

### CODE REVIEW

File: form/create_custom_field_type.rst [hunk 1]
```diff
@@ -267,6 +267,12 @@ link for details), create a ``shipping_widget`` block to handle this:
                 ],
             ]);
 
+    .. deprecated:: 4.3
+
+        The integration of the Templating component in FrameworkBundle has been
+        deprecated since version 4.3 and will be removed in 5.0. Form theming with
+        PHP templates will no longer be supported and you'll need to use Twig instead.
```
Comment (wouterj) [inline form/create_custom_field_type.rst]: I would propose to remove the PHP config example completely. We already only show twig in this article, so it's a bit useless to leave this big code example + note showing a deprecated feature?

**Review by wouterj — CHANGES_REQUESTED** [review]
I agree with all changes except the form theming, which I commented on. Did you have more things you were unsure about?

**Comment by javiereguiluz** [comment]
Now that https://github.com/symfony/symfony/pull/30959 has been merged, this is now ready for the final review.

File: form/create_custom_field_type.rst (hunk 1 above)
Comment (javiereguiluz) [reply form/create_custom_field_type.rst]: OK. Removed it. Thanks.

**Review by Simperfit — APPROVED** [review]

**Review by OskarStark — APPROVED** [review]

**Review by wouterj — APPROVED** [review]
Looks great!

**Review by yceruto — APPROVED** [review]

**Pull request closed without merging**